tests/e2e-playwright/tests/jupyterlab.py

```diff
@@ -0,0 +1,98 @@
+# pylint: disable=redefined-outer-name
+# pylint: disable=unused-argument
+# pylint: disable=unused-variable
+# pylint: disable=too-many-arguments
+# pylint: disable=too-many-statements
+# pylint: disable=unnecessary-lambda
+
+import re
+from http import HTTPStatus
+from typing import Final
+
+from playwright.sync_api import APIRequestContext, Page
+from pydantic import AnyUrl
+from tenacity import Retrying
+from tenacity.retry import retry_if_exception_type
+from tenacity.stop import stop_after_attempt
+from tenacity.wait import wait_fixed
+
+projects_uuid_pattern: Final[re.Pattern] = re.compile(
+    r"/projects/([0-9a-fA-F]{8}-[0-9a-fA-F]{4}-[0-9a-fA-F]{4}-[0-9a-fA-F]{4}-[0-9a-fA-F]{12})"
+)
+
+
+def on_web_socket(ws) -> None:
+    print(f"WebSocket opened: {ws.url}")
+    ws.on("framesent", lambda payload: print(payload))
```

Comment:
TIP: add perhaps prefix to distinguish sent from received. I guess this is equivalent to the little arrows we have 

```python
ws.on("framesent", lambda payload: print("⬇️", payload))
ws.on("framereceived", lambda payload: print("⬆️", payload))
```
:-)

Reply:
I like it! good idea thanks :)